### PR TITLE
Support for Inferenece with no explanation 

### PR DESCRIPTION
**Pull request by owos**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -8,7 +8,6 @@ psutil==5.9.8
 pynvml==11.5.0
 rouge==1.0.1
 sentencepiece==0.2.0
-torch==2.0.0
```
Comment (tobiolatunji) [inline requirements.txt:11]: Why was this removed?

File: scripts/run_no_explanation.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+num_few_shot_values=(0)
+
+for q_type in "${question_types[@]}"; do
+    prompt_file_path="prompts/${prompt_type}_${q_type}${explanation}.txt"
+    echo $prompt_file_path
+    for num_few_shot in "${num_few_shot_values[@]}"; do
+        python bin/main_predictions.py \
+            --data_path $data_path \
+            --prompt_file_path $prompt_file_path \
+            --explanation $explanation \
```
Comment (tobiolatunji) [inline scripts/run_no_explanation.sh:21]: Ideally, this should be a boolean flag, like store true

File: scripts/run_no_explanation.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+pretrained_model_path="microsoft/Phi-3-mini-128k-instruct"
+
+#data_path="data/MedQA-USMLE-4-options-test.csv" #medqa
+data_path="data/afri_med_qa_10k_v1_1_phase_1.csv" #afrimed-qa
+
+question_types=("mcq")
+prompt_type=base
+explanation=_no_exp
```
Comment (tobiolatunji) [inline scripts/run_no_explanation.sh:10]: this seems a bit odd since it's meant to be a boolean option or command line argument.

File: src/utils/utils.py [hunk 4]
```diff
@@ -18,6 +18,7 @@ def parse_arguments():
     parser.add_argument("--pretrained_model_path", type=str, required=True)
     parser.add_argument("--data_path", type=str, required=True)
     parser.add_argument("--prompt_file_path", type=str, default=None)
+    parser.add_argument("--explanation", type=str, default=None)
```
Comment (tobiolatunji) [inline src/utils/utils.py:21]: This should be store true not a string argument

File: scripts/run.sh [hunk 5]
```diff
@@ -5,20 +5,21 @@ pretrained_model_path="gpt-4o"
 data_path="data/MedQA-USMLE-4-options-test.csv" #medqa
 # data_path="data/afri_med_qa_10k_v1_1_phase_1.csv" #afrimed-qa
 
-question_types=("mcq" "saq" "consumer_queries" )
-prompt_type=base
+question_types=("mcq" "saq" "consumer_queries")
+explanation=""
```
Comment (tobiolatunji) [inline scripts/run.sh:9]: This seems odd.

File: src/evals/evaluate.py [hunk 6]
```diff
@@ -2,10 +2,19 @@
...
+def compute_score(q_type, explanation, data):
     valid_data = data[data["rationale"].str.len() > 4].copy()
     if len(valid_data) < 2:
         return (data, 0,0,0,0,0,0,0)
+    
+    if q_type == "mcq":
+        data["correct"] = data["answer"] == data["preds"]
+        accuracy = data["correct"].mean()
+        print("Accuracy:", accuracy)
+        if explanation == "_no_exp":
```
Comment (tobiolatunji) [inline src/evals/evaluate.py:14]: Should simply be `if explanation:`

File: src/utils/utils.py [hunk 7]
```diff
@@ -75,9 +76,10 @@ def logging_cuda_memory_usage():
 def write_results(data, args, score):
     prompt_type = args.prompt_file_path.split("/")[-1].split("_")[0]
     q_type = args.q_type.split('_')[0]
+    explanation = str(args.explanation)
```
Comment (tobiolatunji) [inline src/utils/utils.py:79]: You can do `explanation = '_no_exp' if args.explanation else ""`

**Review by tobiolatunji — CHANGES_REQUESTED** [review]
Great job here.
Minor fix and this should be good to go.
Thanks a million!

File: requirements.txt (hunk 1 above)
Comment (owos) [reply requirements.txt:11]: it is already stated in the repo for people to install their own version of torch. e.g for me now. I only use torch 2.0.0 on the Intron single GPU machine.

**Review by tobiolatunji — APPROVED** [review]
